### PR TITLE
Express popcount, reverse_bits as function prototypes

### PR DESCRIPTION
**Pull request by dneto0**
Builds on #947 

It may be that this is easier to read.

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -2688,6 +2688,38 @@ the contents of the existing table.
   <tr><td>is_normal(float) -&gt; bool<td>OpIsNormal
 </table>
 
+## Integer built-in functions ## {#integer-builtin-functions}
+
+<table class='data'>
+  <thead>
+    <tr><td>Precondition<td>Built-in<td>Description
+  </thead>
+  <tr algorithm="scalar popcount"><td>
```
Comment (kvark) [inline wgsl/index.bs:2697]: I find `popcount` confusing, and also - slightly inconsistent (i.e. we have "reverse_bits" with underscore between words but not "population_count" or something like this). Would it be possible to consider other names, maybe "count_bits()` or something similar?

Comment (jdashg) [reply wgsl/index.bs:2697]: count_ones is a name I like, given other names like count_leading_zeros in other areas.

**Comment by grorg** [comment]
Discussed at the [2020-08-11 meeting](https://docs.google.com/document/d/1YcpsnMS7ioeaJi4qPN1rBnYDgyQxEcudjUfyKmmq4Ik/). 

Resolution: Editors will decide whether or not to merge this in.

**Comment by litherum** [comment]
We should merge this.

File: wgsl/index.bs [hunk 2]
```diff
@@ -2688,6 +2688,38 @@ the contents of the existing table.
...
+          (SPIR-V OpBitCount)
+  <tr algorithm="scalar bit reversal"><td>
+          |T| is u32 or i32<br>
+      <td class="nowrap">`reverse_bits(`|e|`:` |T| `) ->`  |T|
+      <td>Reverses the bits in |e|:  The bit at position |k| of the result equals the
+          bit at position 31-|k| of |e|.<br>
+          (SPIR-V OpBitReverse)
+  <tr algorithm="vector bit reversal"><td>
+          |T| is u32 or i32
+      <td class="nowrap">`reverse_bits(`|e|`:` vec|N|<|T|> `) ->` vec|N|<|T|><br>
```
Comment (dj2) [inline wgsl/index.bs:2717]: Can you update this to reverseBits?

File: wgsl/index.bs [hunk 3]
```diff
@@ -2688,6 +2688,38 @@ the contents of the existing table.
...
 
+## Integer built-in functions ## {#integer-builtin-functions}
+
+<table class='data'>
+  <thead>
+    <tr><td>Precondition<td>Built-in<td>Description
+  </thead>
+  <tr algorithm="scalar popcount"><td>
+          |T| is u32 or i32<br>
+      <td class="nowrap">`popcount(`|e|`:` |T| `) ->` |T|
```
Comment (dj2) [inline wgsl/index.bs:2699]: popCount? popcount, not sure ....

File: wgsl/index.bs (hunk 1 above)
Comment (dneto0) [reply wgsl/index.bs:2697]: GLSL has `bitCount`

I like countOneBits, which goes well with reverseBits. I'll change to that.

File: wgsl/index.bs (hunk 2 above)
Comment (dneto0) [reply wgsl/index.bs]: Done

File: wgsl/index.bs (hunk 3 above)
Comment (dneto0) [reply wgsl/index.bs]: renamed to countOneBits

**Comment by dneto0** [comment]
Rebased.  PTAL editors. @dj2, @litherum

**Pull request merged**